proposals/NNNN-option-set-macro.md

```diff
@@ -78,36 +78,43 @@ The macro would then generate the remaining option set implementation, similar t
 The above declaration would expand out to the following code:
 
 ```swift
+struct ShippingOptions {
+  typealias RawValue = Int
 
- struct ShippingOptions {
-   typealias RawValue = Int
-
-   var rawValue: RawValue
+  var rawValue: RawValue
 
-   init() { self.rawValue = 0 }
+  init() { self.rawValue = 0 }
 
-   init(rawValue: RawValue) { self.rawValue = rawValue }
+  init(rawValue: RawValue) { self.rawValue = rawValue }
 
-   static let nextDay: Self =
-	   Self(rawValue: 1 << Options.nextDay.rawValue)
+  static let nextDay: Self =
+    Self(rawValue: 1 << Options.nextDay.rawValue)
 
-   static let secondDay: Self =
-	   Self(rawValue: 1 << Options.secondDay.rawValue)
+  static let secondDay: Self =
+    Self(rawValue: 1 << Options.secondDay.rawValue)
 
-   static let priority: Self =
-	   Self(rawValue: 1 << Options.priority.rawValue)
+  static let priority: Self =
+    Self(rawValue: 1 << Options.priority.rawValue)
 
-   static let standard: Self =
-	   Self(rawValue: 1 << Options.standard.rawValue)
+  static let standard: Self =
+    Self(rawValue: 1 << Options.standard.rawValue)
 
-	private enum Options: Int {
-		case nextDay, secondDay, priority, standard
-	}
+  private enum Options: Int {
+    case nextDay, secondDay, priority, standard
+  }
 }
 
 extension ShippingOptions: OptionSet { }
 ```
 
+This will require the addition of the `@OptionSet` macro declaration to the standard library:
+
+```swift
+@attached(member)
+@attached(conformance)
+public macro OptionSet<RawType>() = #externalMacro(module: "MacroExamplesPlugin", type: "OptionSetMacro")
```

Comment:
This will be used as `@OptionSet<UInt8>` or similar. If we want to infer the raw value type, we'll need to do something differently. And I think we'll want a defaulted parameter that provides the name of the `Options` enum?